### ISSUE TITLE
Sizes error during data generation for turn model

### ISSUE THREAD

**Issue by codicef** [happypepper/DeepHoldem#15]
Trying to generate data for model 3 I get the following error.

Here is the traceback

f@desktop ~/D/D/Source> th DataGeneration/main_data_generation.lua 3
Generating data ...	
terminal time: 0.6830141544342	
5hTh9s8h 1 10	
NN information:	
epoch	200	
gpu	true	
valid_loss	0.0042649123140391	
NN architecture:	
nn.Sequential {
  [input -> (1) -> (2) -> (3) -> output]
  (1): nn.ConcatTable {
    input
      |`-> (1): nn.Sequential {
      |      [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> (8) -> (9) -> (10) -> output]
      |      (1): nn.Linear(1009 -> 500)
      |      (2): nn.BatchNormalization (2D) (500)
      |      (3): nn.PReLU
      |      (4): nn.Linear(500 -> 500)
      |      (5): nn.BatchNormalization (2D) (500)
      |      (6): nn.PReLU
      |      (7): nn.Linear(500 -> 500)
      |      (8): nn.BatchNormalization (2D) (500)
      |      (9): nn.PReLU
      |      (10): nn.Linear(500 -> 1008)
      |    }
       `-> (2): nn.Sequential {
             [input -> (1) -> output]
             (1): nn.Narrow
           }
       ... -> output
  }
  (2): nn.ConcatTable {
    input
      |`-> (1): nn.Sequential {
      |      [input -> (1) -> output]
      |      (1): nn.SelectTable(1)
      |    }
       `-> (2): nn.Sequential {
             [input -> (1) -> (2) -> (3) -> output]
             (1): nn.DotProduct
             (2): nn.Replicate
             (3): nn.MulConstant
           }
       ... -> output
  }
  (3): nn.CAddTable
}
nextround init_bucket time: 0.52907490730286	
build time: 1.0953109264374	
/home/francesco/torch/install/bin/lua: ./Lookahead/lookahead.lua:189: bad argument #1 to 'copy' (sizes do not match at /home/francesco/torch/extra/cutorch/lib/THC/THCTensorCopy.cu:31)
stack traceback:
	[C]: in function 'copy'
	./Lookahead/lookahead.lua:189: in function '_compute_terminal_equities_terminal_equity'
	./Lookahead/lookahead.lua:326: in function '_compute_terminal_equities'
	./Lookahead/lookahead.lua:93: in function '_compute'
	./Lookahead/lookahead.lua:47: in function 'resolve_first_node'
	./Lookahead/resolving.lua:59: in function 'resolve_first_node'
	./DataGeneration/data_generation.lua:132: in function 'generate_data_file'
	./DataGeneration/data_generation.lua:24: in function 'generate_data'
	DataGeneration/main_data_generation.lua:14: in main chunk
	[C]: in function 'dofile'
	...esco/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: in ?

**Comment by codicef** [comment]
Using CPU model I get the following error

/home/francesco/torch/install/bin/lua: ./Lookahead/lookahead.lua:189: inconsistent tensor size, expected tensor [] and src [1 x 10 x 2 x 1326] to have the same number of elements, but got 0 and 26520 elements respectively at /home/francesco/torch/pkg/torch/lib/TH/generic/THTensorCopy.c:86
stack traceback:
	[C]: in function 'copy'
	./Lookahead/lookahead.lua:189: in function '_compute_terminal_equities_terminal_equity'
	./Lookahead/lookahead.lua:326: in function '_compute_terminal_equities'
	./Lookahead/lookahead.lua:93: in function '_compute'
	./Lookahead/lookahead.lua:47: in function 'resolve_first_node'
	./Lookahead/resolving.lua:59: in function 'resolve_first_node'
	./DataGeneration/data_generation.lua:132: in function 'generate_data_file'
	./DataGeneration/data_generation.lua:24: in function 'generate_data'
	DataGeneration/main_data_generation.lua:14: in main chunk
	[C]: in function 'dofile'
	...esco/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: in ?
